src/core/server/http/router/request.ts

```diff
@@ -97,6 +104,11 @@ export class KibanaRequest<Params = unknown, Query = unknown, Body = unknown> {
 
   public readonly url: Url;
   public readonly route: RecursiveReadonly<KibanaRequestRoute>;
+  /**
+   * This property will be removed in future version of this class, please
+   * use the `getFilteredHeaders` method instead
```

Comment:
I think `@deprecated` is more suitable for such things. Could we create an issue to refactor proxy implementation so we can track this effort and remove the deprecated property?